src/commands/Minion/alch.ts

```diff
@@ -26,11 +26,94 @@ const unlimitedFireRuneProviders = resolveItems([
 	'Tome of fire'
 ]);
 
+export function calculateAlchItemByTripDuration(params: {
+	user: KlasaUser;
+	duration: number;
+	items?: { item: Item; qty: number }[];
+	qtyLimit?: number;
+	calculateByRunesOwned?: boolean;
+}) {
+	let { items, duration, user, qtyLimit, calculateByRunesOwned } = params;
+	const userBank = user.bank().clone();
+
+	if (!items) items = [];
+
+	if (items.length === 0) {
+		for (const i of user.getUserFavAlchs() as Item[]) {
+			// Ignore alchs that are already on the alch items
+			if (items.find(a => a.item.id === i.id)) continue;
```

Comment:
This code block is only run if `items.length === 0`, so this shouldn't be here unless getUserFavAlchs() can return duplicate items, in which case that should probably be fixed.